File: aula3go/Ex2.go
package main

import "fmt"

func main() {
	var arr1 = [10]int{}
	var maior = 0
	var menor = 999999
	var soma = 0
	var media = 0
	var maiores = 0
	var x = 0

	for i := 0; i < 10; i++ {
		fmt.Print("digite o valor para colocar no vetor: ")
		fmt.Scan(&x)

		if x > maior {
			maior = x
		}
		if x < menor {
			menor = x
		}
		if x >= 5 {
			maiores++
		}
		soma += x

		arr1[i] = x
	}

	media = soma / len(arr1)

	fmt.Println("O maior valor é igual a: ", maior)
	fmt.Println("O menor valor é igual a: ", menor)
	fmt.Println("A soma dos valores é igual a: ", soma)
	fmt.Println("A media dos valores é igual a: ", media)
	fmt.Println("O numero de valores maior ou igual a 5 é igual a: ", maiores)

}

File: Aula3java/src/Ex2.java
import java.util.ArrayList;
import java.util.Scanner;

public class Ex2 {
    ArrayList<Integer> valores = new ArrayList<Integer>();
    Scanner sc = new Scanner(System.in);

    int maior = 0;
    int menor = Integer.MAX_VALUE;
    int soma = 0;
    int media = 0;
    int maiores = 0;

    public void Vetor(){

        for (int i = 0; i <= 10; i++){
            System.out.print("Digite um valor para adicionar ao vetor: ");
            int x = sc.nextInt();
            System.out.println("");
            valores.add(x);
        }

    }

    public void Consulta(){
        for(int valor : valores ){
            soma += valor;
            if (valor > maior){
                maior = valor;
            }
            if (valor < menor){
                menor = valor;
            }
            if (valor >= 5){
                maiores++;
            }
        }
        media = soma / valores.size();
    }

    public void Mostrar(){
        Vetor();
        Consulta();
        System.out.println("O maior valor é igual a: " + maior + ". O menor valor é igual a: " + menor + ". A soma de todos os valores é igual a: " + soma + " e a media é igual a: " + media + ". o numero de valores maiores ou iguais a 5 é igual a: " + maiores);
    }
}
